EIPS/eip-4762.md

```diff
@@ -89,25 +89,50 @@ Where `tree_key` and `sub_key` are computed as `tree_key, sub_key = get_storage_
 
 #### Access events for code
 
-In the conditions below, “chunk chunk_id is accessed” is understood to mean an access event of the form
+In the conditions below, _chunk `chunk_id` is accessed_ is understood to mean an access event of the form `(address, (chunk_id + 128) // 256, (chunk_id + 128) % 256)`.
 
-```
-(address, (chunk_id + 128) // 256, (chunk_id + 128) % 256)
-```
+At each step of EVM execution, the corresponding code chunk for the current accessed bytecode is considered accessed:
+
+* For legacy code, corresponds to `PC` (global program counter).
+* For EOFv1 containers, corresponds to the bytecode by referenced by `PC` within `current_section_index`.
+
+In particular, note the following cases:
+
+* The destination of a `JUMP` or positively evaluated `JUMPI` is considered to be accessed, even if the destination is not a jumpdest or is inside pushdata.
+* The destination of a `RJUMP`, `RJUMPV`, `RETF`, or positively evaluated `RJUMPI` is considered to be accessed.
+* The destination of a `JUMPI`, or `RJUMPI` is not considered to be accessed if the jump conditional is `false`.
+* The destination of a jump is not considered to be accessed if the execution gets to the jump opcode but does not have enough gas to pay for the gas cost of executing it (including chunk access cost if the jump is the first opcode in a not-yet-accessed chunk).
+* For legacy code, the destination of a jump is not considered to be accessed if it is beyond the code (`destination >= len(code)`).
+* If code stops execution by walking past the end of the code, `PC = len(code)` is not considered to be accessed.
+
+If the current step of EVM execution is a `PUSH{n}`, all chunks `(PC // CHUNK_SIZE) <= chunk_index <= ((PC + n) // CHUNK_SIZE)` of the callee are accessed, where `PC` is the global program counter for legacy code or `PC` within `current_section_index` for EOFv1 containers.
+
+If a nonzero-read-size `CODECOPY` or `EXTCODECOPY` targeting an account with legacy code read bytes `x...y` inclusive, all chunks `(x // CHUNK_SIZE) <= chunk_index <= (min(y, code_size - 1) // CHUNK_SIZE)` of the accessed contract are accessed:
+
+* Example 1: for a `CODECOPY` with start position 100, read size 50, `code_size = 200`, `x = 100` and `y = 149`
+* Example 2: for a `CODECOPY` with start position 600, read size 0, no chunks are accessed
+* Example 3: for a `CODECOPY` with start position 1500, read size 2000, `code_size = 3100`, `x = 1500` and `y = 3099`
+
+ If `EXTCODECOPY` targets an EOFv1 container, no code chunks are accessed since the code has hardcoded value `EF00`:
+
+Note that `CODESIZE`, `EXTCODESIZE` and `EXTCODEHASH` do NOT access any chunks.
```

Comment:
`EXTCODESIZE` indeed doesn't access any new code-chunk for EOFv1 so no need to clarify.